src/dmd/backend/cgcod.d

```diff
@@ -698,11 +698,7 @@ tryagain:
 
     assert(global87.stackused == 0);             /* nobody in 8087 stack         */
 
-    /* Clean up ndp save array  */
-    mem_free(global87.save);
-    global87.save = null;
-    global87.savetop = 0;
-    global87.savemax = 0;
+    global87.save.__dtor();       // clean up ndp save array
```

Comment:
Manually calling `__dtor` ?
you could put the content of the dtor in a function which you call from the dtor no?